include/crow/websocket.h

```diff
@@ -157,6 +158,36 @@ namespace crow
                     do_write();
                 });
             }
+            
+            /// Send a binary encoded message in blocking mode.
+            void send_binary_blocking(const std::string& msg, boost::system::error_code& ec) override
+            {
+                auto header = build_header(2, msg.size());
+                std::vector<boost::asio::const_buffer> buffer;
+                buffer.emplace_back(boost::asio::buffer(header));
+                buffer.emplace_back(boost::asio::buffer(msg));
+
+                boost::asio::io_service service;
+                boost::asio::deadline_timer deadline_timer(service);
+                deadline_timer.expires_from_now(boost::posix_time::seconds(3));
+
+                deadline_timer.async_wait([&](const boost::system::error_code& err)
+                {
+                    if(!err)
+                    {
+                        ec = boost::system::errc::make_error_code(boost::system::errc::timed_out);
+                        adaptor_.socket().shutdown(boost::asio::ip::tcp::socket::shutdown_send);
```

Comment:
it would be better to use `adaptor_.showtdown_write()`

```suggestion
                        adaptor_.shutdown_write();
```